frontend/packages/console-app/src/components/cloud-shell/setup/CloudShellAdminSetup.tsx

```diff
@@ -1,71 +1,100 @@
 import * as React from 'react';
+import { Formik } from 'formik';
 import { useTranslation } from 'react-i18next';
 import { connect } from 'react-redux';
+import { K8sModel } from '@console/dynamic-plugin-sdk/src/api/common-types';
+import { k8sGetResource, k8sCreateResource } from '@console/dynamic-plugin-sdk/src/utils/k8s';
+import { SectionHeading } from '@console/internal/components/utils';
 import { LoadError } from '@console/internal/components/utils/status-box';
 import { NamespaceModel } from '@console/internal/models';
-import { K8sKind, k8sCreate, k8sGet } from '@console/internal/module/k8s';
 import {
   newCloudShellWorkSpace,
   createCloudShellResourceName,
   CLOUD_SHELL_PROTECTED_NAMESPACE,
 } from '../cloud-shell-utils';
-import TerminalLoadingBox from '../TerminalLoadingBox';
+import {
+  CloudShellSetupFormData,
+  cloudShellSetupValidationSchema,
+  getCloudShellTimeout,
+} from './cloud-shell-setup-utils';
+import CloudSehellSetupForm from './CloudShellSetupForm';
 
 type Props = {
-  onInitialize: (namespace: string) => void;
-  workspaceModel: K8sKind;
+  onSubmit?: (namespace: string) => void;
+  onCancel?: () => void;
+  workspaceModel: K8sModel;
   operatorNamespace: string;
 };
 
 const CloudShellAdminSetup: React.FunctionComponent<Props> = ({
-  onInitialize,
+  onSubmit,
+  onCancel,
   workspaceModel,
   operatorNamespace,
 }) => {
+  const initialValues: CloudShellSetupFormData = {
+    namespace: CLOUD_SHELL_PROTECTED_NAMESPACE,
+    advancedOptions: {
+      timeout: {
+        limit: null,
+        unit: 'm',
+      },
+    },
+  };
+
   const { t } = useTranslation();
 
   const [initError, setInitError] = React.useState<string>();
```

Comment:
Also if this is old code, I don't see any place where a string was saved here.

```suggestion
  const [initError, setInitError] = React.useState<Error>();
```

Reply:
yeah it's an existing one and `initError` is passed here https://github.com/openshift/console/blob/master/frontend/packages/console-app/src/components/cloud-shell/setup/CloudShellAdminSetup.tsx#L72

`<LoadError message={initError} label={t('console-app~OpenShift command line terminal')} />`

so ideally should keep as a string only and pass `error.message`

WDYT?